src/Controller/ApiV1Controller.php

```diff
@@ -0,0 +1,211 @@
+ * Copyright (C) 2009-2010  HO Ngoc Phuong Trang <tranglich@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * PHP version 5
+ *
+ * @category PHP
+ * @package  Tatoeba
+ * @author   HO Ngoc Phuong Trang <tranglich@gmail.com>
+ * @license  Affero General Public License
+ * @link     https://tatoeba.org
+ */
+namespace App\Controller;
+
+use App\Controller\AppController;
+use App\Form\SentencesSearchForm;
+use App\Model\CurrentUser;
+use App\Model\Table\SentencesTable;
+use App\Lib\LanguagesLib;
+use App\Lib\SphinxClient;
+use App\Lib\Licenses;
+use Cake\Core\Configure;
+use Cake\Database\Expression\QueryExpression;
+use Cake\Event\Event;
+use Cake\Routing\Router;
+use Cake\Utility\Hash;
+use Cake\View\ViewBuilder;
+use Exception;
+
+/**
+ * Controller for Api.
+ *
+ * @category Sentences
+ * @package  Controllers
+ * @author   HO Ngoc Phuong Trang <tranglich@gmail.com>
+ * @license  Affero General Public License
+ * @link     https://tatoeba.org
+ */
+class ApiV1Controller extends AppController
+{
+    public $name = 'Sentences';
+    public $components = array (
+        'LanguageDetection',
+        'CommonSentence',
+        'Permissions',
+    );
+    public $helpers = array(
+        'Sentences',
+        'Menu',
+        'Lists',
+        'SentenceButtons',
+        'Html',
+        'Logs',
+        'Pagination',
+        'Comments',
+        'Languages',
+        'CommonModules'
+    );
+
+    public $uses = array(
+        'Audio',
+        'Sentence',
+        'SentencesSentencesLists',
+        'SentencesList',
+        'User',
+        'UsersLanguages',
+        'Tag',
+        'UsersSentences',
+        'Vocabulary'
+    );
+
+    public function initialize() {
+        parent::initialize();
+
+        $params = $this->request->params;
+        $noCsrfActions = ['edit_sentence', 'change_language'];
+        if (in_array($params['action'], $noCsrfActions)) {
+            $this->components()->unload('Csrf');
+        }
+    }
+
+    /**
+     * Show Json of specified sentence id.
+     * 
+     * @param mixed $id Id of the sentence.
+     * 
+     * @return void
+     */
+    public function sentence($id = null) {
+        // We will handle the output directly, so as not to display the header and footer
+        $this->disableAutoRender();
+
+        if (is_numeric($id)) {
```

Comment:
I don’t think we really this `[is_numeric()](https://www.php.net/is_numeric)` check here. The method should return error when the sentence cannot be found, be it because `$id` is not numeric (e.g. `/api_v1/sentence/hello`) or because the provided number doesn’t match any sentence (e.g. `/api_v1/sentence/99999999999999999`). In both cases, `getSentenceWith()` will return null so all you need to do is to check against this returned value.

Reply:
For my use case, https://github.com/Tatoeba/tatoeba2/issues/2669 I'm fine with just getting a blank back. What would you like returned in case of bad data?

Reply:
The proper way for an API to handle this is to return a 4xx error, for example 400. To do that in CakePHP, just throw the exception `\Cake\Http\Exception\NotFoundException`. For an example, you can have a look at this code in ExportsController:
https://github.com/Tatoeba/tatoeba2/blob/dcbad1e3c9d29ce4c3ab3c08c9a774605c21be7d/src/Controller/ExportsController.php#L50-L73

Reply:
It doesn't seem to find the "NotFoundException()". 
![image](https://user-images.githubusercontent.com/7943077/118849749-53964f00-b8d0-11eb-9d5c-8608fb9378b1.png)

I'm not sure how or where to load it. I looked at ExportsController.php and did "use" of the same items there, but didn't fix things. For reference, my code currently:
![image](https://user-images.githubusercontent.com/7943077/118849892-79bbef00-b8d0-11eb-95fa-c1f3f5bf53a6.png)